src/containers/PaymentMethodsPage/PaymentMethodsPage.js

```diff
@@ -19,11 +26,90 @@ import {
 } from '../../components';
 import { TopbarContainer } from '..';
 import { PaymentMethodsForm } from '../../forms';
+import { createStripeSetupIntent, stripeCustomer, loadData } from './PaymentMethodsPage.duck.js';
 
 import css from './PaymentMethodsPage.css';
 
-export const PaymentMethodsPageComponent = props => {
-  const { currentUser, scrollingDisabled, intl } = props;
+const PaymentMethodsPageComponent = props => {
+  const {
+    currentUser,
+    addPaymentMethodError,
+    deletePaymentMethodError,
+    createStripeCustomerError,
+    handleCardSetupError,
+    onCreateSetupIntent,
+    onHandleCardSetup,
+    onCreateStripeCustomer,
+    onAddPaymentMethod,
+    onUpdatePaymentMethod,
+    onDeletePaymentMethod,
+    fetchStripeCustomer,
+    scrollingDisabled,
+    intl,
+  } = props;
+
+  const handleSubmit = params => {
+    const ensuredCurrentUser = ensureCurrentUser(currentUser);
+    const stripeCustomer = ensuredCurrentUser.stripeCustomer;
+    const hasSavedPaymentMethod = stripeCustomer ? stripeCustomer.defaultPaymentMethod : null;
+
+    const { stripe, card, formValues } = params;
+
+    onCreateSetupIntent().then(setupIntent => {
+      const setupIntentClientSecret =
+        setupIntent && setupIntent.attributes ? setupIntent.attributes.clientSecret : null;
+      const { name, addressLine1, addressLine2, postal, state, city, country } = formValues;
+      const addressMaybe =
+        addressLine1 && postal
+          ? {
+              address: {
+                city: city,
+                country: country,
+                line1: addressLine1,
+                line2: addressLine2,
+                postal_code: postal,
+                state: state,
+              },
+            }
+          : {};
+      const billingDetails = {
+        name,
+        email: ensureCurrentUser(currentUser).attributes.email,
+        ...addressMaybe,
+      };
+
+      const paymentParams = {
+        payment_method_data: {
+          billing_details: billingDetails,
+        },
+      };
+
+      const stripeParams = { stripe, card, setupIntentClientSecret, paymentParams };
+
+      onHandleCardSetup(stripeParams)
+        .then(result => {
+          if (result.error) {
+            return;
```

Comment:
I assume that this is not needed since there's `Promise.reject` handling in onHandleCardSetup thunk function.
This could be left in place, but it would be good to have some comment explaining what happens to errors.